// proto/image_api_grpc_web_pb.js
/**
 * @fileoverview gRPC-Web generated client stub for moapis.file_api
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!



const grpc = {};
grpc.web = require('grpc-web');

const proto = {};
proto.moapis = {};
proto.moapis.file_api = require('./image_api_pb.js');

/**
 * @param {string} hostname
 * @param {?Object} credentials
 * @param {?Object} options
 * @constructor
 * @struct
 * @final
 */
proto.moapis.file_api.ImageServiceClient =
    function(hostname, credentials, options) {
  if (!options) options = {};
  options['format'] = 'text';

  /**
   * @private @const {!grpc.web.GrpcWebClientBase} The client
   */
  this.client_ = new grpc.web.GrpcWebClientBase(options);

  /**
   * @private @const {string} The hostname
   */
  this.hostname_ = hostname;

};


/**
 * @param {string} hostname
 * @param {?Object} credentials
 * @param {?Object} options
 * @constructor
 * @struct
 * @final
 */
proto.moapis.file_api.ImageServicePromiseClient =
    function(hostname, credentials, options) {
  if (!options) options = {};
  options['format'] = 'text';

  /**
   * @private @const {!grpc.web.GrpcWebClientBase} The client
   */
  this.client_ = new grpc.web.GrpcWebClientBase(options);

  /**
   * @private @const {string} The hostname
   */
  this.hostname_ = hostname;

};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodDescriptor_ImageService_NewImageResize = new grpc.web.MethodDescriptor(
  '/moapis.file_api.ImageService/NewImageResize',
  grpc.web.MethodType.UNARY,
  proto.moapis.file_api.NewImageRequest,
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodInfo_ImageService_NewImageResize = new grpc.web.AbstractClientBase.MethodInfo(
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.moapis.file_api.NewImageResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.moapis.file_api.NewImageResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.moapis.file_api.ImageServiceClient.prototype.newImageResize =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResize',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResize,
      callback);
};


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.moapis.file_api.NewImageResponse>}
 *     A native promise that resolves to the response
 */
proto.moapis.file_api.ImageServicePromiseClient.prototype.newImageResize =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResize',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResize);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodDescriptor_ImageService_NewImagePreserve = new grpc.web.MethodDescriptor(
  '/moapis.file_api.ImageService/NewImagePreserve',
  grpc.web.MethodType.UNARY,
  proto.moapis.file_api.NewImageRequest,
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodInfo_ImageService_NewImagePreserve = new grpc.web.AbstractClientBase.MethodInfo(
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.moapis.file_api.NewImageResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.moapis.file_api.NewImageResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.moapis.file_api.ImageServiceClient.prototype.newImagePreserve =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImagePreserve',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImagePreserve,
      callback);
};


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.moapis.file_api.NewImageResponse>}
 *     A native promise that resolves to the response
 */
proto.moapis.file_api.ImageServicePromiseClient.prototype.newImagePreserve =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImagePreserve',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImagePreserve);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodDescriptor_ImageService_NewImageResizeAndPreserve = new grpc.web.MethodDescriptor(
  '/moapis.file_api.ImageService/NewImageResizeAndPreserve',
  grpc.web.MethodType.UNARY,
  proto.moapis.file_api.NewImageRequest,
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodInfo_ImageService_NewImageResizeAndPreserve = new grpc.web.AbstractClientBase.MethodInfo(
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.moapis.file_api.NewImageResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.moapis.file_api.NewImageResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.moapis.file_api.ImageServiceClient.prototype.newImageResizeAndPreserve =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResizeAndPreserve',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResizeAndPreserve,
      callback);
};


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.moapis.file_api.NewImageResponse>}
 *     A native promise that resolves to the response
 */
proto.moapis.file_api.ImageServicePromiseClient.prototype.newImageResizeAndPreserve =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResizeAndPreserve',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResizeAndPreserve);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodDescriptor_ImageService_NewImageResizeAtDimensions = new grpc.web.MethodDescriptor(
  '/moapis.file_api.ImageService/NewImageResizeAtDimensions',
  grpc.web.MethodType.UNARY,
  proto.moapis.file_api.NewImageRequest,
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.moapis.file_api.NewImageRequest,
 *   !proto.moapis.file_api.NewImageResponse>}
 */
const methodInfo_ImageService_NewImageResizeAtDimensions = new grpc.web.AbstractClientBase.MethodInfo(
  proto.moapis.file_api.NewImageResponse,
  /**
   * @param {!proto.moapis.file_api.NewImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.NewImageResponse.deserializeBinary
);


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.moapis.file_api.NewImageResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.moapis.file_api.NewImageResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.moapis.file_api.ImageServiceClient.prototype.newImageResizeAtDimensions =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResizeAtDimensions',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResizeAtDimensions,
      callback);
};


/**
 * @param {!proto.moapis.file_api.NewImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.moapis.file_api.NewImageResponse>}
 *     A native promise that resolves to the response
 */
proto.moapis.file_api.ImageServicePromiseClient.prototype.newImageResizeAtDimensions =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/moapis.file_api.ImageService/NewImageResizeAtDimensions',
      request,
      metadata || {},
      methodDescriptor_ImageService_NewImageResizeAtDimensions);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.moapis.file_api.RemoveImageRequest,
 *   !proto.moapis.file_api.RemoveImageResponse>}
 */
const methodDescriptor_ImageService_RemoveImage = new grpc.web.MethodDescriptor(
  '/moapis.file_api.ImageService/RemoveImage',
  grpc.web.MethodType.UNARY,
  proto.moapis.file_api.RemoveImageRequest,
  proto.moapis.file_api.RemoveImageResponse,
  /**
   * @param {!proto.moapis.file_api.RemoveImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.RemoveImageResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.moapis.file_api.RemoveImageRequest,
 *   !proto.moapis.file_api.RemoveImageResponse>}
 */
const methodInfo_ImageService_RemoveImage = new grpc.web.AbstractClientBase.MethodInfo(
  proto.moapis.file_api.RemoveImageResponse,
  /**
   * @param {!proto.moapis.file_api.RemoveImageRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  proto.moapis.file_api.RemoveImageResponse.deserializeBinary
);


/**
 * @param {!proto.moapis.file_api.RemoveImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.moapis.file_api.RemoveImageResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.moapis.file_api.RemoveImageResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.moapis.file_api.ImageServiceClient.prototype.removeImage =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/moapis.file_api.ImageService/RemoveImage',
      request,
      metadata || {},
      methodDescriptor_ImageService_RemoveImage,
      callback);
};


/**
 * @param {!proto.moapis.file_api.RemoveImageRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.moapis.file_api.RemoveImageResponse>}
 *     A native promise that resolves to the response
 */
proto.moapis.file_api.ImageServicePromiseClient.prototype.removeImage =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/moapis.file_api.ImageService/RemoveImage',
      request,
      metadata || {},
      methodDescriptor_ImageService_RemoveImage);
};


module.exports = proto.moapis.file_api;
