// proto/image_api_pb.js
// source: image_api.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.moapis.file_api.ImageDimensions', null, global);
goog.exportSymbol('proto.moapis.file_api.NewImageRequest', null, global);
goog.exportSymbol('proto.moapis.file_api.NewImageResponse', null, global);
goog.exportSymbol('proto.moapis.file_api.NewImageResponseStruct', null, global);
goog.exportSymbol('proto.moapis.file_api.RemoveImageRequest', null, global);
goog.exportSymbol('proto.moapis.file_api.RemoveImageResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.ImageDimensions = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.moapis.file_api.ImageDimensions, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.ImageDimensions.displayName = 'proto.moapis.file_api.ImageDimensions';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.NewImageRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.moapis.file_api.NewImageRequest.repeatedFields_, null);
};
goog.inherits(proto.moapis.file_api.NewImageRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.NewImageRequest.displayName = 'proto.moapis.file_api.NewImageRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.NewImageResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.moapis.file_api.NewImageResponse.repeatedFields_, null);
};
goog.inherits(proto.moapis.file_api.NewImageResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.NewImageResponse.displayName = 'proto.moapis.file_api.NewImageResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.RemoveImageRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.moapis.file_api.RemoveImageRequest.repeatedFields_, null);
};
goog.inherits(proto.moapis.file_api.RemoveImageRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.RemoveImageRequest.displayName = 'proto.moapis.file_api.RemoveImageRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.RemoveImageResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.moapis.file_api.RemoveImageResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.RemoveImageResponse.displayName = 'proto.moapis.file_api.RemoveImageResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.moapis.file_api.NewImageResponseStruct = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.moapis.file_api.NewImageResponseStruct, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.moapis.file_api.NewImageResponseStruct.displayName = 'proto.moapis.file_api.NewImageResponseStruct';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.ImageDimensions.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.ImageDimensions.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.ImageDimensions} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.ImageDimensions.toObject = function(includeInstance, msg) {
  var f, obj = {
    width: jspb.Message.getFieldWithDefault(msg, 1, 0),
    height: jspb.Message.getFieldWithDefault(msg, 2, 0),
    alpha: jspb.Message.getFieldWithDefault(msg, 3, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.ImageDimensions}
 */
proto.moapis.file_api.ImageDimensions.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.ImageDimensions;
  return proto.moapis.file_api.ImageDimensions.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.ImageDimensions} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.ImageDimensions}
 */
proto.moapis.file_api.ImageDimensions.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setWidth(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setHeight(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setAlpha(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.ImageDimensions.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.ImageDimensions.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.ImageDimensions} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.ImageDimensions.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getWidth();
  if (f !== 0) {
    writer.writeUint32(
      1,
      f
    );
  }
  f = message.getHeight();
  if (f !== 0) {
    writer.writeUint32(
      2,
      f
    );
  }
  f = message.getAlpha();
  if (f !== 0) {
    writer.writeUint32(
      3,
      f
    );
  }
};


/**
 * optional uint32 width = 1;
 * @return {number}
 */
proto.moapis.file_api.ImageDimensions.prototype.getWidth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.moapis.file_api.ImageDimensions} returns this
 */
proto.moapis.file_api.ImageDimensions.prototype.setWidth = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint32 height = 2;
 * @return {number}
 */
proto.moapis.file_api.ImageDimensions.prototype.getHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.moapis.file_api.ImageDimensions} returns this
 */
proto.moapis.file_api.ImageDimensions.prototype.setHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional uint32 alpha = 3;
 * @return {number}
 */
proto.moapis.file_api.ImageDimensions.prototype.getAlpha = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.moapis.file_api.ImageDimensions} returns this
 */
proto.moapis.file_api.ImageDimensions.prototype.setAlpha = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.moapis.file_api.NewImageRequest.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.NewImageRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.NewImageRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.NewImageRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    imageList: msg.getImageList_asB64(),
    dimensions: (f = msg.getDimensions()) && proto.moapis.file_api.ImageDimensions.toObject(includeInstance, f),
    tkn: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.NewImageRequest}
 */
proto.moapis.file_api.NewImageRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.NewImageRequest;
  return proto.moapis.file_api.NewImageRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.NewImageRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.NewImageRequest}
 */
proto.moapis.file_api.NewImageRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.addImage(value);
      break;
    case 2:
      var value = new proto.moapis.file_api.ImageDimensions;
      reader.readMessage(value,proto.moapis.file_api.ImageDimensions.deserializeBinaryFromReader);
      msg.setDimensions(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setTkn(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.NewImageRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.NewImageRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.NewImageRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getImageList_asU8();
  if (f.length > 0) {
    writer.writeRepeatedBytes(
      1,
      f
    );
  }
  f = message.getDimensions();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.moapis.file_api.ImageDimensions.serializeBinaryToWriter
    );
  }
  f = message.getTkn();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * repeated bytes image = 1;
 * @return {!Array<string>}
 */
proto.moapis.file_api.NewImageRequest.prototype.getImageList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * repeated bytes image = 1;
 * This is a type-conversion wrapper around `getImageList()`
 * @return {!Array<string>}
 */
proto.moapis.file_api.NewImageRequest.prototype.getImageList_asB64 = function() {
  return /** @type {!Array<string>} */ (jspb.Message.bytesListAsB64(
      this.getImageList()));
};


/**
 * repeated bytes image = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getImageList()`
 * @return {!Array<!Uint8Array>}
 */
proto.moapis.file_api.NewImageRequest.prototype.getImageList_asU8 = function() {
  return /** @type {!Array<!Uint8Array>} */ (jspb.Message.bytesListAsU8(
      this.getImageList()));
};


/**
 * @param {!(Array<!Uint8Array>|Array<string>)} value
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
 */
proto.moapis.file_api.NewImageRequest.prototype.setImageList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {!(string|Uint8Array)} value
 * @param {number=} opt_index
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
 */
proto.moapis.file_api.NewImageRequest.prototype.addImage = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
 */
proto.moapis.file_api.NewImageRequest.prototype.clearImageList = function() {
  return this.setImageList([]);
};


/**
 * optional ImageDimensions dimensions = 2;
 * @return {?proto.moapis.file_api.ImageDimensions}
 */
proto.moapis.file_api.NewImageRequest.prototype.getDimensions = function() {
  return /** @type{?proto.moapis.file_api.ImageDimensions} */ (
    jspb.Message.getWrapperField(this, proto.moapis.file_api.ImageDimensions, 2));
};


/**
 * @param {?proto.moapis.file_api.ImageDimensions|undefined} value
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
*/
proto.moapis.file_api.NewImageRequest.prototype.setDimensions = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
 */
proto.moapis.file_api.NewImageRequest.prototype.clearDimensions = function() {
  return this.setDimensions(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.moapis.file_api.NewImageRequest.prototype.hasDimensions = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string tkn = 3;
 * @return {string}
 */
proto.moapis.file_api.NewImageRequest.prototype.getTkn = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.moapis.file_api.NewImageRequest} returns this
 */
proto.moapis.file_api.NewImageRequest.prototype.setTkn = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.moapis.file_api.NewImageResponse.repeatedFields_ = [1,2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.NewImageResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.NewImageResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.NewImageResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    linkList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    structureList: jspb.Message.toObjectList(msg.getStructureList(),
    proto.moapis.file_api.NewImageResponseStruct.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.NewImageResponse}
 */
proto.moapis.file_api.NewImageResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.NewImageResponse;
  return proto.moapis.file_api.NewImageResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.NewImageResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.NewImageResponse}
 */
proto.moapis.file_api.NewImageResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addLink(value);
      break;
    case 2:
      var value = new proto.moapis.file_api.NewImageResponseStruct;
      reader.readMessage(value,proto.moapis.file_api.NewImageResponseStruct.deserializeBinaryFromReader);
      msg.addStructure(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.NewImageResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.NewImageResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.NewImageResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getLinkList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
  f = message.getStructureList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.moapis.file_api.NewImageResponseStruct.serializeBinaryToWriter
    );
  }
};


/**
 * repeated string link = 1;
 * @return {!Array<string>}
 */
proto.moapis.file_api.NewImageResponse.prototype.getLinkList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.moapis.file_api.NewImageResponse} returns this
 */
proto.moapis.file_api.NewImageResponse.prototype.setLinkList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.moapis.file_api.NewImageResponse} returns this
 */
proto.moapis.file_api.NewImageResponse.prototype.addLink = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.moapis.file_api.NewImageResponse} returns this
 */
proto.moapis.file_api.NewImageResponse.prototype.clearLinkList = function() {
  return this.setLinkList([]);
};


/**
 * repeated NewImageResponseStruct Structure = 2;
 * @return {!Array<!proto.moapis.file_api.NewImageResponseStruct>}
 */
proto.moapis.file_api.NewImageResponse.prototype.getStructureList = function() {
  return /** @type{!Array<!proto.moapis.file_api.NewImageResponseStruct>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.moapis.file_api.NewImageResponseStruct, 2));
};


/**
 * @param {!Array<!proto.moapis.file_api.NewImageResponseStruct>} value
 * @return {!proto.moapis.file_api.NewImageResponse} returns this
*/
proto.moapis.file_api.NewImageResponse.prototype.setStructureList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.moapis.file_api.NewImageResponseStruct=} opt_value
 * @param {number=} opt_index
 * @return {!proto.moapis.file_api.NewImageResponseStruct}
 */
proto.moapis.file_api.NewImageResponse.prototype.addStructure = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.moapis.file_api.NewImageResponseStruct, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.moapis.file_api.NewImageResponse} returns this
 */
proto.moapis.file_api.NewImageResponse.prototype.clearStructureList = function() {
  return this.setStructureList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.moapis.file_api.RemoveImageRequest.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.RemoveImageRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.RemoveImageRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.RemoveImageRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.RemoveImageRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    linkList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    tkn: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.RemoveImageRequest}
 */
proto.moapis.file_api.RemoveImageRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.RemoveImageRequest;
  return proto.moapis.file_api.RemoveImageRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.RemoveImageRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.RemoveImageRequest}
 */
proto.moapis.file_api.RemoveImageRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addLink(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setTkn(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.RemoveImageRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.RemoveImageRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.RemoveImageRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.RemoveImageRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getLinkList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
  f = message.getTkn();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * repeated string link = 1;
 * @return {!Array<string>}
 */
proto.moapis.file_api.RemoveImageRequest.prototype.getLinkList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.moapis.file_api.RemoveImageRequest} returns this
 */
proto.moapis.file_api.RemoveImageRequest.prototype.setLinkList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.moapis.file_api.RemoveImageRequest} returns this
 */
proto.moapis.file_api.RemoveImageRequest.prototype.addLink = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.moapis.file_api.RemoveImageRequest} returns this
 */
proto.moapis.file_api.RemoveImageRequest.prototype.clearLinkList = function() {
  return this.setLinkList([]);
};


/**
 * optional string tkn = 2;
 * @return {string}
 */
proto.moapis.file_api.RemoveImageRequest.prototype.getTkn = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.moapis.file_api.RemoveImageRequest} returns this
 */
proto.moapis.file_api.RemoveImageRequest.prototype.setTkn = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.RemoveImageResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.RemoveImageResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.RemoveImageResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.RemoveImageResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    status: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.RemoveImageResponse}
 */
proto.moapis.file_api.RemoveImageResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.RemoveImageResponse;
  return proto.moapis.file_api.RemoveImageResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.RemoveImageResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.RemoveImageResponse}
 */
proto.moapis.file_api.RemoveImageResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.RemoveImageResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.RemoveImageResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.RemoveImageResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.RemoveImageResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStatus();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string status = 1;
 * @return {string}
 */
proto.moapis.file_api.RemoveImageResponse.prototype.getStatus = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.moapis.file_api.RemoveImageResponse} returns this
 */
proto.moapis.file_api.RemoveImageResponse.prototype.setStatus = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.toObject = function(opt_includeInstance) {
  return proto.moapis.file_api.NewImageResponseStruct.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.moapis.file_api.NewImageResponseStruct} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageResponseStruct.toObject = function(includeInstance, msg) {
  var f, obj = {
    originallink: jspb.Message.getFieldWithDefault(msg, 1, ""),
    resizedlink: jspb.Message.getFieldWithDefault(msg, 2, ""),
    originalid: jspb.Message.getFieldWithDefault(msg, 3, 0),
    resizedid: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.moapis.file_api.NewImageResponseStruct}
 */
proto.moapis.file_api.NewImageResponseStruct.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.moapis.file_api.NewImageResponseStruct;
  return proto.moapis.file_api.NewImageResponseStruct.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.moapis.file_api.NewImageResponseStruct} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.moapis.file_api.NewImageResponseStruct}
 */
proto.moapis.file_api.NewImageResponseStruct.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setOriginallink(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setResizedlink(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setOriginalid(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setResizedid(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.moapis.file_api.NewImageResponseStruct.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.moapis.file_api.NewImageResponseStruct} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.moapis.file_api.NewImageResponseStruct.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOriginallink();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getResizedlink();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getOriginalid();
  if (f !== 0) {
    writer.writeUint32(
      3,
      f
    );
  }
  f = message.getResizedid();
  if (f !== 0) {
    writer.writeUint32(
      4,
      f
    );
  }
};


/**
 * optional string originalLink = 1;
 * @return {string}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.getOriginallink = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.moapis.file_api.NewImageResponseStruct} returns this
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.setOriginallink = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string resizedLink = 2;
 * @return {string}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.getResizedlink = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.moapis.file_api.NewImageResponseStruct} returns this
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.setResizedlink = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional uint32 originalID = 3;
 * @return {number}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.getOriginalid = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.moapis.file_api.NewImageResponseStruct} returns this
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.setOriginalid = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional uint32 resizedID = 4;
 * @return {number}
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.getResizedid = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.moapis.file_api.NewImageResponseStruct} returns this
 */
proto.moapis.file_api.NewImageResponseStruct.prototype.setResizedid = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


goog.object.extend(exports, proto.moapis.file_api);
